### PR TITLE
feat: Add Device S7 generator script

### PR DESCRIPTION
**Pull request by jiekechoo**
**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-compose/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: compose-builder/add-device-s7.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#  * or implied. See the License for the specific language governing permissions and limitations under
+#  * the License.
+#  *
+#  *******************************************************************************/
+
+version: '3.7'
+
+services:
+  device-s7:
+    image: ${DEVICE_SVC_REPOSITORY}/device-s7${ARCH}:${DEVICE_MODBUS_VERSION}
```
Comment (lenny-intel) [inline compose-builder/add-device-s7.yml:20]: Add DEVICE_S7_VERSION to `.env` file
```suggestion
    image: ${DEVICE_SVC_REPOSITORY}/device-s7${ARCH}:${DEVICE_S7_VERSION}
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: compose-builder/add-device-s7.yml (hunk 1 above)
Comment (jiekechoo) [reply compose-builder/add-device-s7.yml]: Thanks for your help. :D

**Review by lenny-intel — APPROVED** [review]

**Comment by lenny-intel** [comment]
@jiekechoo , please fix your last commit message (amend) causing the Semantic PR check to fail.

**Comment by lenny-intel** [comment]
@cherrycl , This PR doesn't impact any smoke tests, but all are failing, Please advise @jiekechoo  on how to resolve it.

**Comment by cherrycl** [comment]
@jiekechoo Please rebase and squash commits.

**Comment by jiekechoo** [comment]
Another PR replace this.

**Pull request closed without merging**